### ISSUE TITLE
'icalderivedproperty.h' file not found: missing explicit dependency?

### ISSUE THREAD

**Issue by ajacoutot** [libical/libical#321]
Hi.

Building libical 3.0.0 there seems to be a missing dependency ordering. It randomly fails for me with the following error:
```
===>  Building for libical-3.0.0
[1/239] cd /hack/objdirs/pobj/libical-3.0.0/build-amd64/src/libicalss && /usr/local/bin/cmake -DTOPS:FILEPATH=/hack/objdirs/pobj/libical-3.0.0/libical-3.0.0 -DTOPB:FILEPATH=/hack/objdirs/pobj/libical-3.0.0/build-amd64 -DICAL_FILE_H_FILE:FILEPATH=/hack/objdirs/pobj/libical-3.0.0/build-amd64/src/libicalss/icalss.h -DBDB_FOUND=1 -P /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libicalss/icalss_file.cmake
[2/239] cd /hack/objdirs/pobj/libical-3.0.0/build-amd64/src/libical && /usr/bin/perl -I /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/scripts/ /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/scripts//mkderivedparameters.pl -i /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libical/icalderivedparameter.c.in -c /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/design-data/parameters.csv > /hack/objdirs/pobj/libical-3.0.0/build-amd64/src/libical/icalderivedparameter.c
[3/239] cd /hack/objdirs/pobj/libical-3.0.0/build-amd64/src/libical && /usr/bin/perl -I/hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/scripts/ /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/scripts//mkderivedvalues.pl -i /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libical/icalderivedvalue.c.in -c /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/design-data/value-types.csv > /hack/objdirs/pobj/libical-3.0.0/build-amd64/src/libical/icalderivedvalue.c
[4/239] /hack/objdirs/pobj/libical-3.0.0/bin/cc -DHAVE_CONFIG_H -DWITH_CXX_BINDINGS -Dlibical_icalss_EXPORTS -I. -Isrc -I/hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src -Isrc/libical -I/hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libical -I/usr/local/include/db4 -O2 -pipe  -I/usr/local/include -fvisibility=hidden -Wno-deprecated -Wall -Wno-unknown-pragmas -Wextra -Winit-self -Wmissing-include-dirs -Wunused -Wno-div-by-zero -Wundef -Wpointer-arith -Wtype-limits -Wwrite-strings -Werror=return-type -Wsizeof-pointer-memaccess -Wformat -Wformat-security -Wredundant-decls -Qunused-arguments -DNDEBUG -MD -MT src/libicalss/CMakeFiles/icalss-static.dir/icalfileset.c.o -MF src/libicalss/CMakeFiles/icalss-static.dir/icalfileset.c.o.d -o src/libicalss/CMakeFiles/icalss-static.dir/icalfileset.c.o   -c /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libicalss/icalfileset.c
FAILED: src/libicalss/CMakeFiles/icalss-static.dir/icalfileset.c.o 
/hack/objdirs/pobj/libical-3.0.0/bin/cc -DHAVE_CONFIG_H -DWITH_CXX_BINDINGS -Dlibical_icalss_EXPORTS -I. -Isrc -I/hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src -Isrc/libical -I/hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libical -I/usr/local/include/db4 -O2 -pipe  -I/usr/local/include -fvisibility=hidden -Wno-deprecated -Wall -Wno-unknown-pragmas -Wextra -Winit-self -Wmissing-include-dirs -Wunused -Wno-div-by-zero -Wundef -Wpointer-arith -Wtype-limits -Wwrite-strings -Werror=return-type -Wsizeof-pointer-memaccess -Wformat -Wformat-security -Wredundant-decls -Qunused-arguments -DNDEBUG -MD -MT src/libicalss/CMakeFiles/icalss-static.dir/icalfileset.c.o -MF src/libicalss/CMakeFiles/icalss-static.dir/icalfileset.c.o.d -o src/libicalss/CMakeFiles/icalss-static.dir/icalfileset.c.o   -c /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libicalss/icalfileset.c
In file included from /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libicalss/icalfileset.c:26:
In file included from /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libicalss/icalfileset.h:26:
In file included from /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libicalss/icalcluster.h:23:
In file included from /hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libical/icalcomponent.h:25:
/hack/objdirs/pobj/libical-3.0.0/libical-3.0.0/src/libical/icalproperty.h:24:10: fatal error: 'icalderivedproperty.h' file not found
#include "icalderivedproperty.h"        /* To get icalproperty_kind enumerations */
         ^~~~~~~~~~~~~~~~~~~~~~~
1 error generated.
ninja: build stopped: subcommand failed.
*** Error 1 in . (/hack/cvs/ports/devel/cmake/cmake.port.mk:31 'do-build': @cd /hack/objdirs/pobj/libical-3.0.0/build-amd64 && exec /usr/bin...)
*** Error 1 in . (/hack/cvs/ports/infrastructure/mk/bsd.port.mk:2724 '/hack/objdirs/pobj/libical-3.0.0/build-amd64/.build_done')
*** Error 1 in . (/hack/cvs/ports/infrastructure/mk/bsd.port.mk:1945 '/hack/cvs/ports/packages/amd64/all/libical-3.0.0p2.tgz')
*** Error 1 in . (/hack/cvs/ports/infrastructure/mk/bsd.port.mk:2446 '_internal-package')
*** Error 1 in /hack/cvs/ports/textproc/libical (/hack/cvs/ports/infrastructure/mk/bsd.port.mk:2425 'package')
```

**Comment by ajacoutot** [comment]
I am using this patch for the time being which does the trick for me. I don't know cmake that much so I am not realy sure whether this is the ideal thing to do or not (hence no pull request).
Thanks.

```
--- src/libicalss/CMakeLists.txt.orig
+++ src/libicalss/CMakeLists.txt
@@ -78,10 +78,10 @@ add_custom_command(
 )
 
 add_library(icalss ${LIBRARY_TYPE} ${icalss_LIB_SRCS})
-add_dependencies(icalss icalss-header)
+add_dependencies(icalss ical-header icalss-header)
 if(NOT SHARED_ONLY)
   add_library(icalss-static STATIC ${icalss_LIB_SRCS})
-  add_dependencies(icalss-static icalss-header)
+  add_dependencies(icalss-static ical-header icalss-header)
 endif()
 
 target_link_libraries(icalss ical)
@@ -134,10 +134,10 @@ if(WITH_CXX_BINDINGS)
     )
   endif()
   add_library(icalss_cxx ${LIBRARY_TYPE} ${icalsscxx_LIB_SRCS})
-  add_dependencies(icalss_cxx icalss-header)
+  add_dependencies(icalss_cxx ical-header icalss-header)
   if(NOT SHARED_ONLY)
     add_library(icalss_cxx-static STATIC ${icalsscxx_LIB_SRCS})
-    add_dependencies(icalss_cxx-static icalss-header)
+    add_dependencies(icalss_cxx-static ical-header icalss-header)
   endif()
 
   target_link_libraries(icalss_cxx icalss ical_cxx ${CMAKE_THREAD_LIBS_INIT})
```

**Comment by dilyanpalauzov** [comment]
Duplicate of #317.

**Comment by winterz** [comment]
thanks for the patch
fixed with commit 2f5f57552cc43eb56b654ba152bff648ada3d369
